tests/languages/docker_test.py

```diff
@@ -62,6 +62,25 @@ def invalid_attribute():
         assert docker.get_docker_user() == ()
 
 
+@pytest.mark.parametrize(
+    ('info_ret', 'expect_root'),
+    (
+        ((0, b'["name=rootless","name=cgroupns"]', b''), True),
+        ((0, b'["name=cgroupns"]', b''), False),
+        ((1, b'', b''), False),
+    ),
+)
+def test_docker_user_rootless_docker(info_ret, expect_root):
+    docker._is_rootless_docker.cache_clear()
```

Comment:
:thinking: maybe we should clear this before _and_ after the test run: otherwise we risk polluting following tests with whatever value we last injected

Reply:
usually a better approach is something like this:

```python
@pytest.fixture(autouse=True)
def _avoid_cache():
    with mock.patch.object(docker, '_is_rootless_docker', docker._is_rootless_docker.__wrapped__):
        yield
```

this bypasses the cache for the duration of the test